### ISSUE TITLE
Alien::Build::Plugin::Fetch::CurlCommand> curl: option -J: is unknown

### ISSUE THREAD

**Issue by Chayyoo** [Perl5-Alien/Alien-Build#147]
Trying to install Alien::FFI on Centos6 gives the above error and fails (line 203).
(tried both cpan -i and cpanm).

**Comment by plicease** [comment]
@Chayyoo will take a look at this when I get a chance.  You may be able to install `Net::SSLeay`, `IO::Socket::SSL` to force the `HTTP::Tiny` plugin to work around this.

**Comment by Chayyoo** [comment]
Thanks for the tip, but installing Net::SSLeay fails as well...

**Comment by plicease** [comment]
```
"/usr/bin/perl" -MAlien::Build::MM=cmd -e download
Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine
Alien::Build::Plugin::Fetch::CurlCommand> writeout: ab-filename     :%{filename_effective}\n
Alien::Build::Plugin::Fetch::CurlCommand> writeout: ab-content_type :%{content_type}\n
Alien::Build::Plugin::Fetch::CurlCommand> writeout: ab-url          :%{url_effective}\n
Alien::Build::Plugin::Fetch::CurlCommand> + /usr/bin/curl -L -f -O -J -w @writeout https://sourceware.org/pub/libffi/
Alien::Build::Plugin::Fetch::CurlCommand> curl: option -J: is unknown
Alien::Build::Plugin::Fetch::CurlCommand> curl: try 'curl --help' or 'curl --manual' for more information
error in curl fetch at /usr/local/share/perl5/Alien/Build/Plugin/Fetch/CurlCommand.pm line 203.
make: *** [_alien/mm/download] Error 2
FAIL
! Installing Alien::FFI failed. See /root/.cpanm/work/1575479985.12085/build.log for details. Retry with --force to force install it.
```

**Comment by plicease** [comment]
@Chayyoo lol this is why we try curl first.  You can also use the system libffi if you install `libffi-devel`

**Comment by Chayyoo** [comment]
I did it: for Net::SSLeay I first needed openssl-devel. After that everything went smoothly.
Thanks!

**Comment by plicease** [comment]
CentOS 6 curl also has a problem with the filename of a URL with a trailing `/`:

```
[root@bf016e796750 tmp]# /usr/bin/curl -L -f -O -w @writeout https://sourceware.org/pub/libffi/
curl: Remote file name has no length!
curl: try 'curl --help' or 'curl --manual' for more information
```

and 

```
[root@bf016e796750 tmp]# /usr/bin/curl -L -f -o x -w @writeout https://sourceware.org/pub/libffi/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 14754    0 14754    0     0  44660      0 --:--:-- --:--:-- --:--:--  138k
curl: unknown --write-out variable: 'filename_effective'
ab-filename     :
ab-content_type :text/html;charset=UTF-8
ab-url          :https://sourceware.org/pub/libffi/
```

**Comment by plicease** [comment]
> I did it: for Net::SSLeay I first needed openssl-devel. After that everything went smoothly.
> Thanks!

Great!  Thanks for reporting this I'd like to get the curl plugin working for CentOS 6, so we will leave this open.

**Comment by plicease** [comment]
#148 adds a probe for the -J option and skips curl for now (released as 1.93).  Ideally we'd work with curls that don't have the -J option, and may do so in the future so I've opened #149 specifically to add support for the older curls.  This addresses (to some extent) CentOS 6, so I am closing this issue.
